### PR TITLE
Arc6

### PR DESCRIPTION
**Pull request by chaen**
This was tested here https://gitlab.cern.ch/CLICdp/iLCDirac/diracos-test/-/pipelines/2435576 . The integration tests fail for reasons totally independent from ARC. Now we need someone to actually test that against ARC CEs @fstagni @andresailer @VladimirRomanovsky  @rajanandakumar  ? 

BEGINRELEASENOTES

CHANGE: Move to ARC 6

ENDRELEASENOTES

### CODE REVIEW

**Comment by petricm** [comment]
I have uploaded the build of this to the standard location [https://diracos.web.cern.ch/diracos/releases/diracos-arc6.tar.gz](https://diracos.web.cern.ch/diracos/releases/diracos-arc6.tar.gz)

**Comment by andresailer** [comment]
I can push this to the iLCDirac test server, maybe even today...

**Comment by andresailer** [comment]
I am getting
```
2021-03-26 10:55:11 UTC WorkloadManagement/SiteDirector/arc-ce02.gridpp.rl.ac.uk WARN: Failed to submit job because  SUBMITTER_PLUGIN_NOT_LOADED : ARC library installation problem?
```
Am I missing some setting somewhere? bashrc comes from puppet

**Comment by chaen** [comment]
I have absolutely no idea :-) 
There are arc variables defined in `diracosrc`, maybe they need an update ?

**Comment by andresailer** [comment]
Are there additional plugins needed? @rajanandakumar 
https://www.nordugrid.org/arc/arc6/users/client_install.html#additional-plugins

**Comment by rajanandakumar** [comment]
@andresailer how do I reproduce this issue? Thanks!

**Comment by andresailer** [comment]
I use the diracos-arc6 tarball and run the sitedirector for an ARC CE as normal.
What I don't see in that tarbal is something like the ftpjob lib. But I am not sure which plugin is used for the job submission at the moment? Arc rest? gridftpjob?

**Comment by rajanandakumar** [comment]
I think that we use the gridftpjob plugin as in

https://github.com/DIRACGrid/DIRAC/blob/e0c0b1a94f3007ecc5e83f4bfd2964844a28defb/src/DIRAC/Resources/Computing/ARCComputingElement.py#L227

**Comment by chaen** [comment]
It would seem that the packaging changed and the split a lot of things. 

So this is the complete list of packages we produced:

```
u'nordugrid-arc-devel': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-devel-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'python2-nordugrid-arc': u'/var/lib/mock/epel-6-x86_64/result/python2-nordugrid-arc-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-test-utils': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-test-utils-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-arcctl': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-arcctl-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-arcctl-service': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-arcctl-service-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-gridftpd': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-gridftpd-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-nordugridmap': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-nordugridmap-6.10.1-1.el6.py27.usc4.noarch.rpm'
u'nordugrid-arc-hed': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-hed-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-internal': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-internal-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-client': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-client-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-arcrest': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-arcrest-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-archery-manage': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-archery-manage-6.10.1-1.el6.py27.usc4.noarch.rpm'
u'nordugrid-arc-monitor': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-monitor-6.10.1-1.el6.py27.usc4.noarch.rpm'
u'nordugrid-arc-plugins-globus-common': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-globus-common-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-xrootd': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-xrootd-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-infosys-ldap': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-infosys-ldap-6.10.1-1.el6.py27.usc4.noarch.rpm'
u'nordugrid-arc-plugins-gridftp': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gridftp-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-python': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-python-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-arex': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-arex-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-globus': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-globus-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-datadelivery-service': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-datadelivery-service-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-gfal': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gfal-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-needed': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-needed-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-lcas-lcmaps': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-lcas-lcmaps-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-wn': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-wn-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-debuginfo': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-debuginfo-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-community-rtes': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-community-rtes-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-gridftpjob': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gridftpjob-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'nordugrid-arc-plugins-s3': u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-s3-6.10.1-1.el6.py27.usc4.x86_64.rpm'}

```

Those that we shipped with DIRACOS are 

```

u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-client-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-needed-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-globus-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'/var/lib/mock/epel-6-x86_64/result/python2-nordugrid-arc-6.10.1-1.el6.py27.usc4.x86_64.rpm'
u'/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-6.10.1-1.el6.py27.usc4.src.rpm'
```

Please let me know which one you want to add

**Comment by rajanandakumar** [comment]
After staring at it for some time, I think that the following should keep us going (I am not able to see the pipeline in the original message above for the error)

nordugrid-arc
nordugrid-arc-client
nordugrid-arc-plugins-needed
nordugrid-arc-plugins-globus
nordugrid-arc-plugins-arcrest
nordugrid-arc-plugins-gridftpjob
nordugrid-arc-plugins-globus-common
nordugrid-arc-plugins-gridftp
nordugrid-arc-plugins-xrootd
nordugrid-arc-plugins-gfal
python2-nordugrid-arc

**Comment by chaen** [comment]
So much ?!?! How come ? :-) 

I've recompiled by hand and kept all the RPMs. 

@andresailer would you please be so kind as to take `/afs/cern.ch/user/c/chaen/public/arc6/allArc.tar.gz` and to extract the RPMs @rajanandakumar listed in the DIRACOS of your test machine ? you can do that by being in the diracos folder, and running `rpm2cpio <rpmfile> | cpio -dvim`

**Comment by andresailer** [comment]
I have extracted the plugins as instructed, but I still don't get pilots.

```
  346  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-client-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  347  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-needed-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  348  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-globus-common-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  349  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-globus-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  350  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-arcrest-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  351  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gridftp-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  352  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gridftpjob-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  353  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-xrootd-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  354  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-gfal-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  355  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/python2-nordugrid-arc-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
  359  rpm2cpio allarcs/var/lib/mock/epel-6-x86_64/result/nordugrid-arc-plugins-python-6.10.1-1.el6.py27.usc4.x86_64.rpm | cpio -dvim
```
Also nordugrid-arc-plugins-python, just in case.

I don't get the ARC_PLUGIN_NOT_LOADED warning any longer, but
```
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector VERBOSE: Evaluating queue kek2-ce01.cc.kek.jp_nordugrid-lsf-gridilc_long
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector INFO: kek2-ce01.cc.kek.jp_nordugrid-lsf-gridilc_long: Slots=100, TQ jobs(pilotsWeMayWantToSubmit)=3, Pilots: waiting 0, to submit=3
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector VERBOSE: Getting pilot proxy for /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sailer/CN=683529/CN=Andre Sailer/ilc_pilot 345600 long
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector INFO: Going to submit pilots (a maximum of 3 pilots to kek2-ce01.cc.kek.jp_nordugrid-lsf-gridilc_long queue)
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector VERBOSE: pilotOptions: -S ILC-Development -l ILCDIRAC -r v31r0-pre -C dips://voilcdiractest71.cern.ch:9135/Configuration/Server -e ILC -N kek2-ce01.cc.kek.jp -Q nordugrid-lsf-gridilc_long -n LCG.KEK.jp -o /Resources/Computing/CEDefaults/SubmitPool=gLite
2021-04-06 09:25:09 UTC WorkloadManagement/SiteDirector/kek2-ce01.cc.kek.jp VERBOSE: Executable file path: /opt/dirac/data/work/SiteDirector/DIRAC__eLX8L_pilotwrapper.py
2021-04-06 09:25:13 UTC WorkloadManagement/SiteDirector/kek2-ce01.cc.kek.jp WARN: Failed to submit job because  Job not submitted - incorrect job description? (missing field in XRSL string?)
2021-04-06 09:25:13 UTC WorkloadManagement/SiteDirector/kek2-ce01.cc.kek.jp WARN: Failed to submit job because  some error from the CE - possibly CE problems?
2021-04-06 09:25:13 UTC WorkloadManagement/SiteDirector/kek2-ce01.cc.kek.jp WARN: Failed to submit job because  ... maybe above messages will give a hint.
2021-04-06 09:25:13 UTC WorkloadManagement/SiteDirector ERROR: Failed submission to queue Queue kek2-ce01.cc.kek.jp_nordugrid-lsf-gridilc_long:
, No pilot references obtained from the ARC job submission
```

With debug enabled

```
2021-04-06 09:34:02 UTC WorkloadManagement/SiteDirector INFO: Going to submit pilots (a maximum of 1 pilots to arc-ce02.gridpp.rl.ac.uk_nordugrid-Condor-grid3000M queue)
2021-04-06 09:34:02 UTC WorkloadManagement/SiteDirector VERBOSE: pilotOptions: -S ILC-Development -l ILCDIRAC -r v31r0-pre -C dips://voilcdiractest71.cern.ch:9135/Configuration/Server -e ILC -N arc-ce02.gridpp.rl.ac.uk -Q nordugrid-Condor-grid3000M -n LCG.RAL-LCG2.uk -o /Resources/Computing/CEDefaults/SubmitPool=gLi
te
2021-04-06 09:34:02 UTC WorkloadManagement/SiteDirector/arc-ce02.gridpp.rl.ac.uk VERBOSE: Executable file path: /opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: String successfully parsed as nordugrid:xrsl.
VERBOSE: SendCommand: Response: 250 "jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m" is current directory
VERBOSE: Generating nordugrid:xrsl job description output
VERBOSE: SendCommand: Response: 229 Entering Extended Passive Mode (|||50486|)
VERBOSE: FTP Job Control: Data channel: 130.246.180.247:50486
VERBOSE: Disconnect: Failed aborting - ignoring: Handle not in the proper state
INFO: Transfer from file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py to gsiftp://arc-ce02.gridpp.rl.ac.uk:2811/jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: DataMover: cycle
INFO: Real transfer from file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py to gsiftp://arc-ce02.gridpp.rl.ac.uk:2811/jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Creating buffer: 1048576 x 2
VERBOSE: DataMove::Transfer: no checksum calculation for file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Failed to load plugin for URL (empty)
VERBOSE: Failed to load plugin for URL (empty)
INFO: Using buffered transfer method
INFO: write_thread: get and pass buffers
VERBOSE: Waiting for buffer
FATAL: Failed to execute command
VERBOSE: write_thread: out failed - aborting
VERBOSE: write_thread: exiting
VERBOSE: buffer: read EOF : yes
VERBOSE: buffer: write EOF: yes
VERBOSE: buffer: error    : yes, read: no, write: yes
VERBOSE: Closing read channel
VERBOSE: Closing write channel
VERBOSE: (Re)Trying next destination
VERBOSE: DataMover: cycle
INFO: Real transfer from file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py to gsiftp://arc-ce02.gridpp.rl.ac.uk:2811/jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Creating buffer: 1048576 x 2
VERBOSE: DataMove::Transfer: no checksum calculation for file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Failed to load plugin for URL (empty)
VERBOSE: Failed to load plugin for URL (empty)
INFO: Using buffered transfer method
INFO: write_thread: get and pass buffers
VERBOSE: Waiting for buffer
FATAL: Failed to execute command
VERBOSE: write_thread: out failed - aborting
VERBOSE: write_thread: exiting
VERBOSE: buffer: read EOF : yes
VERBOSE: buffer: write EOF: yes
VERBOSE: buffer: error    : yes, read: no, write: yes
VERBOSE: Closing read channel
VERBOSE: Closing write channel
VERBOSE: (Re)Trying next destination
VERBOSE: DataMover: cycle
INFO: Real transfer from file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py to gsiftp://arc-ce02.gridpp.rl.ac.uk:2811/jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Creating buffer: 1048576 x 2
VERBOSE: DataMove::Transfer: no checksum calculation for file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py
VERBOSE: Failed to load plugin for URL (empty)
VERBOSE: Failed to load plugin for URL (empty)
INFO: Using buffered transfer method
INFO: write_thread: get and pass buffers
VERBOSE: Waiting for buffer
FATAL: Failed to execute command
VERBOSE: write_thread: out failed - aborting
VERBOSE: write_thread: exiting
VERBOSE: buffer: read EOF : yes
VERBOSE: buffer: write EOF: yes
VERBOSE: buffer: error    : yes, read: no, write: yes
VERBOSE: Closing read channel
VERBOSE: Closing write channel
VERBOSE: DataMover: cycle
VERBOSE: DataMover: destination out of tries - exit
ERROR: Failed uploading file file:/opt/dirac/data/work/SiteDirector/DIRAC_ehLllC_pilotwrapper.py to gsiftp://arc-ce02.gridpp.rl.ac.uk:2811;blocksize=1048576;checksum=no/jobs/jMPODm8Gxnync1XDjqYugZkqABFKDmABFKDmOoFLDmABFKDmIGF00m/DIRAC_ehLllC_pilotwrapper.py: Failed while writing to destination
INFO: Submit: Failed uploading local input files
```

**Comment by andresailer** [comment]
Also export
`export ARC_LOCATION=/opt/dirac/pro/diracos/usr`
and for now the LD_LIBRARY_PATH to find 
```
$ find . -name libarcglobusutils.so.3
./usr/lib64/libarcglobusutils.so.3
```
so that this command
`/opt/dirac/pro/diracos/usr/lib64/arc/arc-dmcgridftp`

works to upload the input files for the jobs.

**Comment by chaen** [comment]
New pipeline https://gitlab.cern.ch/CLICdp/iLCDirac/diracos-test/-/pipelines/2463150

**Comment by petricm** [comment]
The build passed but there are missing links
```
./tests/integration/test_ldd.sh
Finding all the binary dependencies, and putting them in /tmp/diracos/missingDependencies.txt
New missing dependencies
libarccommon.so.3
libarccommunication.so.3
libarccompute.so.3
libarccredential.so.3
libarccredentialstore.so.3
libarccrypto.so.3
libarcdata.so.3
libarcdatastaging.so.3
libarcinfosys.so.0
libarcloader.so.3
libarcmessage.so.3
libarcotokens.so.3
libarcsecurity.so.3
libarcwsaddressing.so.0
libarcwssecurity.so.3
libarcxmlsec.so.3
```

I have dumped the tar file `diracos-arc6-2.tar.gz` to the usual location

**Comment by chaen** [comment]
my bad, I reshuffled the list of packages to ship, and I removed one by mistake.... I redo it

**Comment by chaen** [comment]
New pipeline... https://gitlab.cern.ch/CLICdp/iLCDirac/diracos-test/-/pipelines/2465290

**Comment by chaen** [comment]
The build succeeds. @petricm can you please move it as `diracos-arc6` with the md5 such that @andresailer can install it ?

**Comment by petricm** [comment]
I have overwritten the old `diracos-arc6.tar.gz` with the new build

**Comment by andresailer** [comment]
Pulled the new tarball. Pilots are still being submitted

**Comment by chaen** [comment]
so all good ? Want to do more tests or should we merge ?

**Comment by andresailer** [comment]
Pilot status is updated, pilot output can be downloaded. So seems to be OK from my point of view.

**Comment by chaen** [comment]
OK then I squash all the commit and we are good to go

File: diracos/scriptTemplates/diracosrc_tpl.sh [hunk 1]
```diff
@@ -19,8 +19,8 @@ export PATH
 export PYTHONWARNINGS="ignore"
 
 # ARC Computing Element
-ARC_PLUGIN_PATH=$DIRACOS/usr/lib64/arc;
-export ARC_PLUGIN_PATH;
+export ARC_LOCATION=$DIRACOS/usr
```
Comment (petricm) [inline diracos/scriptTemplates/diracosrc_tpl.sh:22]:
```suggestion
ARC_LOCATION=$DIRACOS/usr
```

**Review by petricm — APPROVED** [review]

**Pull request merged**